drivers/uktty/ns16550/ns16550.c

```diff
@@ -142,6 +142,13 @@ static int early_init(struct ukplat_bootinfo *bi)
 
 	ns16550_uart_base = CONFIG_LIBUKTTY_NS16550_EARLY_CONSOLE_BASE;
 
+	/* Configure the port */
+	rc = init_ns16550();
+	if (unlikely(rc)) {
+		uk_pr_err("Could not initialize ns16550 (%d)\n", rc);
```

Comment:
Wait so we print on failing to initialize console?

Reply:
In https://github.com/unikraft/unikraft/pull/1472#discussion_r1694224280 you ask for an error message before the console is initialized, so which one is it? 😉 Either way, we could also state that we make no assumptions of the overall system's state at the library level (e.g. imagine VGA was already on) and print messages on all errors.

Reply:
> you ask

u right i lost sight of that

>  make no assumptions of the overall system's state at the library level

ok sure

Reply:
Agree with Michalis' comment. I imagine that with the `ukprint` rework we will have a kernel message buffer. This would in principle enable that we can print out the buffer buffer later on a different device and would see this message.